### PR TITLE
Add Blockchain Node Engine monitoring dashboard integration

### PR DESCRIPTION
**Pull request by kscroggs**

### CODE REVIEW

File: dashboards/google-blockchain-node-engine/blockchain_node_engine_dashboard.json [hunk 1]
```diff
@@ -0,0 +1,257 @@
+{
+  "displayName": "Blockchain Node Engine Dashboard",
```
Comment (EvanSimpson) [inline dashboards/google-blockchain-node-engine/blockchain_node_engine_dashboard.json:2]: Small nit to remove redundant "Dashboard" from the name here.

```suggestion
  "displayName": "Blockchain Node Engine",
```

File: dashboards/google-blockchain-node-engine/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+sample_dashboards:
+  -
+    category: Blockchain Node Engine
+    id: blockchain_node_engine
```
Comment (EvanSimpson) [inline dashboards/google-blockchain-node-engine/metadata.yaml:4]: This ID needs to match the json/png file names.

```suggestion
    id: blockchain_node_engine_dashboard
```

**Review by EvanSimpson — CHANGES_REQUESTED** [review]

File: dashboards/google-blockchain-node-engine/blockchain_node_engine_dashboard.json [hunk 3]
```diff
@@ -0,0 +1,257 @@
+{
+  "displayName": "Blockchain Node Engine",
+  "gridLayout": {
```
Comment (EvanSimpson) [inline dashboards/google-blockchain-node-engine/blockchain_node_engine_dashboard.json:3]: I believe gridLayout is deprecated in favor of mosaicLayout. I pasted this dashboard into the dashboard builder and it helpfully did the conversion for me. Can we make that change here?

```
{
  "displayName": "Blockchain Node Engine",
  "mosaicLayout": {
    "columns": 12,
    "tiles": [
      {
        "height": 4,
        "widget": {
          "title": "Block Height",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/block_height\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\""
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6
      },
      {
        "height": 4,
        "widget": {
          "title": "Peer Count",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/active_peers\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\""
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "xPos": 6
      },
      {
        "height": 4,
        "widget": {
          "title": "Pending Transactions",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/pending_transactions\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\""
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "yPos": 4
      },
      {
        "height": 4,
        "widget": {
          "title": "Request Count",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "perSeriesAligner": "ALIGN_RATE"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/total_request_count\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\"",
                    "secondaryAggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    }
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "xPos": 6,
        "yPos": 4
      },
      {
        "height": 4,
        "widget": {
          "title": "Memory Utilization",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/memory/utilization\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\""
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "yPos": 8
      },
      {
        "height": 4,
        "widget": {
          "title": "CPU Utilization",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/cpu/utilization\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\""
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "xPos": 6,
        "yPos": 8
      },
      {
        "height": 4,
        "widget": {
          "title": "Disk Operations",
          "xyChart": {
            "chartOptions": {
              "mode": "COLOR"
            },
            "dataSets": [
              {
                "minAlignmentPeriod": "60s",
                "plotType": "LINE",
                "targetAxis": "Y1",
                "timeSeriesQuery": {
                  "timeSeriesFilter": {
                    "aggregation": {
                      "alignmentPeriod": "60s",
                      "perSeriesAligner": "ALIGN_RATE"
                    },
                    "filter": "metric.type=\"blockchainnodeengine.googleapis.com/node/disk/total_ops_count\" resource.type=\"blockchainnodeengine.googleapis.com/BlockchainNode\"",
                    "secondaryAggregation": {
                      "alignmentPeriod": "60s",
                      "crossSeriesReducer": "REDUCE_MEAN",
                      "groupByFields": [
                        "metric.label.\"direction\"",
                        "resource.label.\"blockchain_node_id\""
                      ],
                      "perSeriesAligner": "ALIGN_MEAN"
                    }
                  }
                }
              }
            ],
            "timeshiftDuration": "0s",
            "yAxis": {
              "label": "",
              "scale": "LINEAR"
            }
          }
        },
        "width": 6,
        "yPos": 12
      }
    ]
  }
}
```

Comment (kscroggs) [reply dashboards/google-blockchain-node-engine/blockchain_node_engine_dashboard.json]: Thanks!

**Pull request merged**